src/basic/mkdir.c

```diff
@@ -19,7 +19,8 @@
 int mkdir_safe_internal(
                 const char *path,
                 mode_t mode,
-                uid_t uid, gid_t gid,
+                uid_t uid,
+                gid_t gid,
                 MkdirFlags flags,
                 mkdirat_func_t _mkdirat) {
 
```

Comment:
(we should probably rework this whole function around open_mkdir_at(), i.e. create the dir, and immediately pin it via fd, and then adjust mode/ownership and verify it via the fd)